qabelbox/src/main/java/de/qabel/qabelbox/chat/view/adapters/ChatMessageViewHolderBase.kt

```diff
@@ -0,0 +1,25 @@
+package de.qabel.qabelbox.chat.view.adapters
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import de.qabel.qabelbox.chat.dto.ChatMessage
+import de.qabel.qabelbox.chat.dto.MessagePayload
+import de.qabel.qabelbox.helper.Formatter
+import kotlinx.android.synthetic.main.chat_message_in.view.*
+
+abstract class ChatMessageViewHolderBase<T : MessagePayload>(itemView: View) : RecyclerView.ViewHolder(itemView) {
```

Comment:
If you parametrize the DTO with a (T: MessagePayload), you can remove the unchecked cast and use a safe one.